### ISSUE TITLE
[LightningCLI] Parameters of optimizer and learning-rate scheduler now required by default

### ISSUE THREAD

**Issue by francescomilano172** [PyTorchLightning/pytorch-lightning#9856]
## 🐛 Bug

On the current master branch, `LightningCLI` requires both the optimizer and the learning-rate scheduler parameters to be set. Is this the desired behavior? This was not the case until PyTorch Lightning 1.4.9, and to me this seems undesirable, since:
- One might want to use one's own `configure_optimizers` method, e.g., to specify which parameters the optimizer should optimize. If I'm not mistaken, this is currently not possible,  since a custom `configure_optimizers` is overwritten (cf. https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/cli.py#L719-L723) and all the parameters are automatically set to be optimized (cf. https://github.com/PyTorchLightning/pytorch-lightning/blob/4decbc0d95f9d1f5c3f437fbafb167d6ec3beb40/pytorch_lightning/utilities/cli.py#L713).
- One might to keep a constant learning rate and not use a learning-rate scheduler.

### To Reproduce
With the `BoringModel`:
- File `train.py`:
```python
import pl_bolts
from pytorch_lightning import LightningDataModule, LightningModule
from pytorch_lightning.utilities.cli import LightningCLI
import torch
from torch.utils.data import DataLoader


class BoringModel(LightningModule):

    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)

    def forward(self, x):
        return self.layer(x)

    def loss(self, batch, prediction):
        # An arbitrary loss to have a loss that updates the model weights during `Trainer.fit` calls
        return torch.nn.functional.mse_loss(prediction,
                                            torch.ones_like(prediction))

    def training_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"loss": loss}

    def training_step_end(self, training_step_outputs):
        return training_step_outputs

    def training_epoch_end(self, outputs) -> None:
        torch.stack([x["loss"] for x in outputs]).mean()

    def validation_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"x": loss}

    def validation_epoch_end(self, outputs) -> None:
        torch.stack([x['x'] for x in outputs]).mean()

    def test_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        self.log('fake_test_acc', loss)
        return {"y": loss}

    def test_epoch_end(self, outputs) -> None:
        torch.stack([x["y"] for x in outputs]).mean()

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
        return [optimizer], [lr_scheduler]


class MyDataModule(LightningDataModule):

    def __init__(self):
        super().__init__()
        self.dataset = pl_bolts.datasets.RandomDataset(size=32, num_samples=10)

    def train_dataloader(self):

        return DataLoader(self.dataset, shuffle=True, batch_size=2)


cli = LightningCLI(BoringModel, MyDataModule)
```
- File `config.yaml`:
```yaml
optimizer:
  class_path: torch.optim.Adam
  init_args:
    lr: 0.001
lr_scheduler:
  class_path: torch.optim.lr_scheduler.ExponentialLR
  init_args:
    gamma: 0.1
```
- Use the following command to run: `python train.py fit --config config.yaml`

If the lines with the `lr_scheduler` parameters are commented (similarly with the `optimizer` parameters, e.g., running simply `python train.py fit`), the following error is raised:
```
train.py: error: Configuration check failed :: Parser key "lr_scheduler": Value "{}" does not validate against any of the types in typing.Union[torch.optim.lr_scheduler.CosineAnnealingLR, torch.optim.lr_scheduler.CosineAnnealingWarmRestarts, torch.optim.lr_scheduler.CyclicLR, torch.optim.lr_scheduler.ExponentialLR, torch.optim.lr_scheduler.LambdaLR, torch.optim.lr_scheduler.MultiStepLR, torch.optim.lr_scheduler.MultiplicativeLR, torch.optim.lr_scheduler.OneCycleLR, torch.optim.lr_scheduler.StepLR]:
  - Type <class 'torch.optim.lr_scheduler.CosineAnnealingLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.CosineAnnealingWarmRestarts'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.CyclicLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.ExponentialLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.LambdaLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.MultiStepLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.MultiplicativeLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.OneCycleLR'> expects an str or a Dict with a class_path entry but got "{}"
  - Type <class 'torch.optim.lr_scheduler.StepLR'> expects an str or a Dict with a class_path entry but got "{}"
```

### Expected behavior

No error is raised if the optimizer and learning-rate parameters are not passed to `LightningCLI`. In particular, if the `LightningModule` defines a `configure_optimizers` -- as in the above example -- and the command `python train.py fit` is run, training should run with no errors.

### Environment

- PyTorch Lightning Version (e.g., 1.3.0): 4decbc0d95f9d1f5c3f437fbafb167d6ec3beb40
- LightningBolts Version: 0.4.0
- PyTorch Version (e.g., 1.8): 1.8.0
- Python version: 3.6.9
- OS (e.g., Linux): Ubuntu 18.04
- CUDA/cuDNN version: -
- GPU models and configuration: -
- How you installed PyTorch (`conda`, `pip`, source): pip
- If compiling from source, the output of `torch.__config__.show()`: -
- Any other relevant information: -
